app/js/lib/foxbox/api.js

```diff
@@ -266,78 +266,78 @@ export default class API {
+    const selectors = Array.from(this[p.watchChannels].values()).map(
       ({ id }) => ({ id })
     );
 
     return this.put('channels/get', selectors)
       .then((response) => {
         Object.keys(response).forEach((key) => {
-          const getter = this[p.watchGetters].get(key);
-          if (!getter) {
+          const channel = this[p.watchChannels].get(key);
+          if (!channel) {
             return;
           }
 
-          this[p.updateGetterValue](getter, response[key]);
+          this[p.updateChannelValue](channel, response[key]);
         });
       });
   }
 
   /**
-   * Updates getter value if needed. If value has changed, appropriate event is
+   * Updates channel value if needed. If value has changed, appropriate event is
    * fired.
    *
-   * @param {{ id: string, value: Object }} getter Getter to update value for.
-   * @param {Object} getterValue Getter value returned from the server.
+   * @param {{ id: string, value: * }} channel Channel to update value for.
+   * @param {*} newValue New channel value returned from the server.
    *
    * @private
    */
-  [p.updateGetterValue](getter, getterValue) {
+  [p.updateChannelValue](channel, newValue) {
     let valueChanged = false;
 
-    if (!getterValue || !getter.value) {
-      valueChanged = getterValue !== getter.value;
+    if (!newValue || !channel.value) {
+      valueChanged = newValue !== channel.value;
     } else {
-      const [valueType] = Object.keys(getterValue);
+      const [valueType] = Object.keys(newValue);
       if (valueType === 'Error') {
         console.error(
-          'Failed to retrieve value for getter (%s): %o',
-          getter.id,
-          getterValue[valueType]
+          'Failed to retrieve value for channel (%s): %o',
+          channel.id,
+          newValue[valueType]
         );
 
         return;
       }
 
-      const newValue = getterValue[valueType];
-      const oldValue = getter.value[valueType];
-
-      if (newValue && oldValue && typeof newValue === 'object') {
+      if (newValue && channel.value && typeof channel.value === 'object') {
         // @todo If value is a non-null object, we use their JSON representation
         // to compare values. It's not performant and not reliable at all, but
         // this OK until we have such values, once we support them we should
         // have dedicated utility function for deep comparing objects.
-        valueChanged = JSON.stringify(newValue) !== JSON.stringify(oldValue);
+        valueChanged = JSON.stringify(newValue) !==
+          JSON.stringify(channel.value);
       } else {
-        valueChanged = newValue !== oldValue;
+        valueChanged = newValue !== channel.value;
       }
     }
 
     if (valueChanged) {
-      getter.value = Object.freeze(getterValue);
-      this[p.watchEventBus].emit(getter.id, getter.value);
+      // In ES6, a non-object argument will be treated as if it was a frozen
```

Comment:
No I mean, previously "channel.value" was always object (e.g. `{ OpenClosed: "Open"}`), but now it can be just primitive value (e.g. "Open"). In ES5 mode `Object.freeze("Open")` will be complaining, but in ES6 (like in all our supported browsers), everything will be ok :)